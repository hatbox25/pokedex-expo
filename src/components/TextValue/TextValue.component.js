import React from "react";
import { View, Text } from "react-native";

import config from "./TextValue.config";
import styles from "./TextValue.styles";

const TextValue = ({ label, value }) => {
  return (
    <View style={styles.container}>
      <Text>{label}</Text>
      <Text style={styles.valueText}>{value}</Text>
    </View>
  );
};

TextValue.displayName = config.displayName;
TextValue.defaultProps = config.defaultProps;
TextValue.propTypes = config.propTypes;

export default TextValue;
