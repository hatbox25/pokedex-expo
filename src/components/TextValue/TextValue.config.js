import PropTypes from "prop-types";

const displayName = "TextValue";
const propTypes = {
  label: PropTypes.any.isRequired,
  value: PropTypes.any.isRequired,
};
const defaultProps = {
  label: "label",
  value: "Value",
};

export default { displayName, propTypes, defaultProps };
