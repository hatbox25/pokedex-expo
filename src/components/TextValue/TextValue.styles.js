export default {
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
    width: "100%",
    paddingBottom: 16,
  },

  valueText: {
    fontWeight: "600",
  },
};
