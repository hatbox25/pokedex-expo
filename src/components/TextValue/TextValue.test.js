import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import TextValue from "./TextValue.component";
import config from "./TextValue.config";

const props = {
  ...config.defaultProps,
};

let component;
beforeEach(() => {
  component = shallow(<TextValue {...props} />);
});

afterEach(() => {
  jest.clearAllMocks();
});

describe("TextValue Snap Test", () => {
  test("should render correctly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
