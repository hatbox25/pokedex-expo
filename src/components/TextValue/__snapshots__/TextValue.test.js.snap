// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextValue Snap Test should render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 16,
      "width": "100%",
    }
  }
>
  <Text>
    label
  </Text>
  <Text
    style={
      Object {
        "fontWeight": "600",
      }
    }
  >
    Value
  </Text>
</View>
`;
